samcli/local/lambda_service/service.py

```diff
@@ -0,0 +1,128 @@
+"""Local Lambda Service"""
+
+import logging
+import re
+import io
+import os
+
+from flask import Flask, request
+
+
+from samcli.local.services.base_service import BaseService
+from samcli.local.lambdafn.exceptions import FunctionNotFound
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalLambdaService(BaseService):
+
+    _DEFAULT_PORT = 3001
```

Comment:
I don't like hard-coding defaults like port & host deep within this class. Let's make the port mandatory & have it dependency injected by the caller. The caller could read from a config file or from CLI args.

We did this for the APIGW service but I don't think it is a good idea there either

Reply:
Make sense. I can't recall why I did this initially for APIGW originally. Will work on cleaning this up (and for APIGW as well).